Modix.Bot/Behaviors/PromotionPollBehavior/PromotionDialogueBehavior.cs

```diff
@@ -0,0 +1,161 @@
+                    return;
+
+                if (reaction.Name == _approveEmoji.Name)
+                    ++poll.Approve;
+                else if(reaction.Name == _disproveEmoji.Name)
+                    ++poll.Disprove;
+
+                await poll.Message.ModifyAsync(m => m.Embed = BuildPollEmbed(poll.Campaign, poll.Approve, poll.Disprove).Build());
+            }
+        }
+
+        private async Task CreatePromoPoll(PromotionActionCreationData campaign)
+        {
+            var pollChannel = await GetPollChannel(_discordSocketClient.GetGuild(campaign.GuildId));
+            var message = await pollChannel.SendMessageAsync(embed: BuildPollEmbed(campaign.Campaign, 1, 0).Build());
+
+            await message.AddReactionsAsync(new[] {(IEmote)_approveEmoji, _disproveEmoji});
+
+            var poll = new CachedPromoPoll
+            {
+                Message = message,
+                Campaign = campaign.Campaign,
+                Approve = 1,
+                Disprove = 0
+            };
+
+            _memoryCache.Set(campaign.Campaign.Id, poll);
+            _memoryCache.Set(message.Id, poll);
+        }
+
+        private async Task DeletePromoPoll(PromotionActionCreationData campaign)
+        {
+            if (!_memoryCache.TryGetValue(campaign.Campaign.Id, out CachedPromoPoll poll))
+                return;
+            await poll.Message.DeleteAsync();
+        }
+
+        private EmbedBuilder BuildPollEmbed(PromotionCampaignBrief campaign, int Approve, int Disprove)
+        {
+                var boldRole = $"**{MentionUtils.MentionRole(campaign.TargetRole.Id)}**";
+                var user = _discordSocketClient.GetUser(campaign.Subject.Id);
+
+                ValueTask<Color> aviColor;
+
+                if ((user.GetAvatarUrl(size: 16) ?? user.GetDefaultAvatarUrl()) is { } avatarUrl)
+                {
+                    aviColor = _imageService.GetDominantColorAsync(new Uri(avatarUrl));
+                }
+                else
+                {
+                    aviColor = new ValueTask<Color>(Color.Teal);
+                }
+
+                return new EmbedBuilder()
+                    .WithTitle($"#{campaign.Id}")
+                    .WithDescription($"**{campaign.Subject.Nickname}** Has been nominated for promotion to {boldRole} ")
+                    .WithColor(aviColor.Result)
+                    .WithUserAsAuthor(user)
+                    .AddField("The current vote total stands at: ", $" {Approve} {_approveEmoji} / {Disprove} {_disproveEmoji}")
+                    .WithTimestamp(_utcNow)
+                    .WithFooter("Please react with the appropriate reaction to cast your vote!");
+        }
+
+        private async Task<ITextChannel> GetPollChannel(IGuild guild)
+        {
+            var getPollChannel = await _designatedChannelService
+                .GetDesignatedChannelsAsync(guild, DesignatedChannelType.PromotionPoll);
+            return getPollChannel.First() as ITextChannel;
```

Comment:
Don't assume that any designated channels exist. Maybe staff don't actually want this feature. (MODiX is hosted by more than just the discord-csharp guild).